src/cloudai/report_generator/tool/tensorboard_data_reader.py

```diff
@@ -45,8 +46,8 @@ def extract_data(self, tag: str) -> List[Tuple[int, float]]:
         for root, _, files in os.walk(self.directory_path):
             for file in files:
                 if file.startswith("events.out.tfevents"):
-                    path = os.path.join(root, file)
-                    reader = SummaryReader(path)
+                    path = Path(root) / file
+                    reader = SummaryReader(str(path))
```

Comment:
Any reason not to switch to `Path` for `SummaryReader`?

Reply:
As `SummaryReader` is imported from tbparse, we don't have control over the class.
```
        from tbparse import SummaryReader  # lazy import to improve overall performance

        data = []
        for root, _, files in os.walk(self.directory_path):
            for file in files:
                if file.startswith("events.out.tfevents"):
                    path = Path(root) / file
                    reader = SummaryReader(str(path))
                    df = reader.scalars
                    if tag in df["tag"].values:
                        filtered_data = df[df["tag"] == tag]
                        data.extend(filtered_data[["step", "value"]].values)
        return data
```

Reply:
Ah, sorry, thought it is our code. Thank you!